dashboard/config/initializers/fast_localization.rb

```diff
@@ -13,3 +13,30 @@
   I18n.backend.init_translations if I18n.backend.respond_to? :init_translations
   I18n.t 'hello'
 end
+
+# Patch the I18n FileUpdateChecker to only load changed i18n files when updated.
+Dashboard::Application.config.after_initialize do |app|
+  app.reloaders.each do |reloader|
+    if reloader.is_a?(ActiveSupport::FileUpdateChecker) && reloader.files == I18n.load_path
+      reloader.block = lambda{|files| I18n.backend.load_translations(files)}
+    end
+  end
+end
+
+module UpdateCheckerExt
+  # Expose @block and @files variables.
+  attr_accessor :block, :files
+
+  # Pass list of updated files as a block parameter.
+  def execute
+    @last_watched = watched
```

Comment:
this calls [`#watched`](https://github.com/rails/rails/blob/4-2-stable/activesupport/lib/active_support/file_update_checker.rb#L93-L99) in `ActiveSupport::FileUpdateChecker`.

the `execute` implementation overrides the [existing method](https://github.com/rails/rails/blob/4-2-stable/activesupport/lib/active_support/file_update_checker.rb#L70-L79) with the ability to pass a list of updated files as a block parameter.